src/validations/passport-gated/index.ts

```diff
@@ -27,46 +50,158 @@ function hasValidIssuanceAndExpiration(credential, proposalTs) {
+
+    // Scorer done calculating passport score during submission
+    if (submittedPassport.ok && submissionData.status === 'DONE') {
+      return evalPassportScore(submissionData, scoreThreshold);
+    }
+  }
+
+  // Passport Score was already calculated
+  if (scoreResponse.ok && scoreData.status === 'DONE') {
+    return evalPassportScore(scoreData, scoreThreshold);
+  }
+
+  // Try to fetch Passport Score if still processing (scoreData.status === 'PROCESSING')
+  for (let i = 0; i < PASSPORT_SCORER_MAX_ATTEMPTS; i++) {
+    const scoreResponse = await fetch(GET_PASSPORT_SCORE_URI + currentAddress, {
+      headers
+    });
+    const scoreData = await scoreResponse.json();
+
+    if (scoreResponse.ok && scoreData.status === 'DONE') {
+      return evalPassportScore(scoreData, scoreThreshold);
+    }
+    console.log(
+      `[passport] Waiting for scorer... (${i}/${PASSPORT_SCORER_MAX_ATTEMPTS})`
+    );
+    await snapshot.utils.sleep(3e3);
+  }
+  return false;
+}
+
 export default class extends Validation {
   public id = 'passport-gated';
   public github = 'snapshot-labs';
-  public version = '0.1.0';
+  public version = '1.0.0';
   public title = 'Gitcoin Passport Gated';
   public description =
-    'Protect your proposals from spam and vote manipulation by requiring users to have a Gitcoin Passport.';
+    'Protect your proposals from spam and vote manipulation by requiring users to have a valid Gitcoin Passport.';
 
   async validate(currentAddress = this.author): Promise<boolean> {
     const requiredStamps = this.params.stamps || [];
     const operator = this.params.operator;
+    const scoreThreshold = this.params.scoreThreshold || 0;
     if (!operator) throw new Error('Operator is required');
 
-    const stampsResponse = await fetch(
-      GET_PASSPORT_STAMPS_URI + currentAddress,
-      { headers }
-    );
-
-    const stampsData = await stampsResponse.json();
-
-    if (!stampsData?.items) {
-      console.log('[passport] Unknown error', stampsData);
-      throw new Error('Unknown error');
-    }
-    if (stampsData.items.length === 0) return false;
-
     const provider = snapshot.utils.getProvider(this.network);
     const proposalTs = (await provider.getBlock(this.snapshot)).timestamp;
-    // check expiration for all stamps
-    const validStamps = stampsData.items
-      .filter((stamp) =>
-        hasValidIssuanceAndExpiration(stamp.credential, proposalTs)
-      )
-      .map((stamp) => stamp.credential.credentialSubject.provider);
-
-    if (operator === 'AND') {
-      return requiredStamps.every((stamp) => validStamps.includes(stamp));
-    } else if (operator === 'OR') {
-      return requiredStamps.some((stamp) => validStamps.includes(stamp));
-    }
-    return false;
+    const validStamps = await validateStamps(
+      currentAddress,
+      operator,
+      proposalTs,
+      requiredStamps
+    );
+    const validScore = await validatePassportScore(
+      currentAddress,
+      scoreThreshold
+    );
```

Comment:
HI @santteegt This should not be called if there is no `scoreThreshold`

Reply:
This address `0x24F15402C6Bb870554489b2fd2049A85d75B982f` got ENS stamp but it still returns `false` because of the passport score 🤔